### DISCUSSION TITLE
I don't want to use firebase, I want totally self-hosted experience

### DISCUSSION THREAD

**Discussion by michalroxorpl** [hoppscotch/hoppscotch#1923] [Q&A] [answered]
Hello,
I love hoppscotch, successfully deployed it on premises, managed to make it start / stop / reload with systemd service file, however I can't login using email, as it says API key is wrong. After further reading I've found out hoppscotch uses firebase as it's database backend. Projects I want to use it with are of a totally non-disclosure type, in other words I am/we are not allowed for any data to leak. As we all know what is uploaded to google, looses it's privacy, so I would love to find out an alternative to firebase backend, we just do not want to use any google related cancer software.
Is it possible to use local MySQL or PostgreSQL database as it's backend?

**Comment by liyasthomas** [comment]
We'll be providing self-hostable Hoppscotch instances in near future.

**Reply by abhaychaware** [reply]
Is there any progress on this? I have the same requirement as mentioned by @michalroxorpl above.

**Reply by ghost** [reply]
Hello, I am having the same problem as mentioned above, I try to authenticate but the API key is lost, we have hoppcotch hosted on an ubuntu server and we want to manage the users that are going to use it.

**Reply by dasaintgray** [reply]
> We'll be providing self-hostable Hoppscotch instances in near future.



> We'll be providing self-hostable Hoppscotch instances in near future.

We are happy to read this.

**Reply by Entropy-9973** [reply]
This is the only thing preventing my company from using it right now.
It should get a much higher adoption rate once this is sorted out.
I would recommend just to throw the data in the file system or use sqlite as a quick solution

**Reply by Seenox** [reply]
Should we all probably watch #1817 for status updates on this?

**Reply by jobartim44** [reply]
Maybel @liyasthomas could give news on it ? On January he was saying that it was deferred to Q2-Q3 2022. Now that the v3 is released maybe we could have fresh informations about the espected schedule.

**Comment by jobartim44** [comment]
For information, liyasthomas mention this feature in a reply on another issue : "We’re working towards a self hosted Hoppscotch instance support for Q1 2023"

**Reply by kabelman93** [reply]
are there any updates in this regard?

**Comment by liyasthomas** [comment, marked as answer]
Follow instruction on [documentation](https://docs.hoppscotch.io/) to set up Hoppscotch locally.
